apps/ssr-tests-v9/src/test.ts

```diff
@@ -0,0 +1,93 @@
+import chalk from 'chalk';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Browser } from 'puppeteer';
+
+import { PROVIDER_ID } from './utils/constants';
+import { hrToSeconds } from './utils/helpers';
+import { launchBrowser } from './utils/launchBrowser';
+
+class RenderError extends Error {}
```

Comment:
"inheritance" in JS is a confusing paradigm that doesn't work as one might think it works. with that leveraging `instanceof` is potentialy dangeours operation as it wont give you similar guarantees like in true OO languagues.

for those reason I'd recommend if we could follow one of following patterns:

1. dont use "extends" rather literal objects

```ts
error = { message: message.text(), name: 'RenderError' };


// later on

  if (err.name === 'RenderError') {
```

2. properly instantiate `name` of custom error

```ts
class RenderError extends Error {
  name = 'RangeError'
}

// check will be the same as in 1st option = no instanceof
```


WDYT ?

Reply:
curious, what is wrong with `instanceof` in this case?

Reply:
in this particular case there is nothing wrong. 

**To give you better context:** 

from  experience I avoid "volatile" JS lang features in general, which besides various JS quirks is also "OOP features", thus duck typing is always an obvious choice.

- doesn't work with objects that are shared between windows/frames (not our case)
- problematic with isArray / http://perfectionkills.com/instanceof-considered-harmful-or-how-to-write-a-robust-isarray/
- if target output is es5 (not our case) this will not work